dynamiqs/time_array.py

```diff
@@ -329,36 +336,22 @@ def __add__(self, other: ArrayLike | TimeArray) -> TimeArray:
 
 
 class PWCTimeArray(TimeArray):
-    times: Array  # (nv+1,)
+    times: Array  # (..., nv+1,)
     values: Array  # (..., nv)
-    array: Array  # (n, n)
+    array: Array  # (..., n, n)
```

Comment:
Can we start with the batching only carried by `values` first? Supporting batching on arrays here is a new feature (that needs added documentation, tests, etc...), I'd rather add this in another PR if possible, to limit the amount of change in this PR.

Reply:
It's essentially free to implement, I'll do it in next PR